spring-core/src/main/java/facade/BookingFacadeImpl.java

```diff
@@ -0,0 +1,149 @@
+        return userService.retrieveUserByEmail(email);
+    }
+
+    public List<User> getUsersByName(String name, int pageSize, int pageNum) {
+        checkPageSize(pageSize, pageNum);
+        final List<User> users = userService.retrieveUserByName(name);
+        LOG.debug("Users: " + users);
+        return users.isEmpty() ? Collections.emptyList() : listByPage(users, pageSize, pageNum);
+    }
+
+    public User createUser(User user) {
+        return userService.addUser(user);
+    }
+
+    public User updateUser(User user) {
+        return userService.updateUser(user);
+    }
+
+    public boolean deleteUser(long userId) {
+        return userService.removeUser(userId);
+    }
+
+    public Ticket bookTicket(long userId, long eventId, int place, Ticket.Category category) {
+        List<Ticket> bookedTickets = getBookedTickets(eventId);
+        final Optional<Ticket> bookedTicket = bookedTickets.stream()
+                .filter(ticket -> place == ticket.getPlace() && category.equals(ticket.getCategory()) && userId == ticket.getUserId()).findFirst();
+        if (bookedTicket.isPresent()) {
+            LOG.error("User is already preset: " + bookedTicket.get());
+            throw new IllegalStateException("Place already booked");
+        }
+        Ticket ticket = new Ticket();
+        ticket.setUserId(userId);
+        ticket.setEventId(eventId);
+        ticket.setPlace(place);
+        ticket.setCategory(category);
+        return ticketService.bookTicket(ticket);
+    }
+
+    public List<Ticket> getBookedTickets(User user, int pageSize, int pageNum) {
+        checkPageSize(pageSize, pageNum);
+        List<Ticket> tickets = ticketService.retrieveUserTickets(user.getId());
+        LOG.debug("Tickets: " + tickets);
+        tickets.sort(ticketComparatorByEventDate().reversed());
+        LOG.debug("Sorted tickets: " + tickets);
+        return listByPage(tickets, pageSize, pageNum);
+    }
+
+    public List<Ticket> getBookedTickets(Event event, int pageSize, int pageNum) {
+        checkPageSize(pageSize, pageNum);
+        List<Ticket> tickets = getBookedTickets(event.getId());
+        LOG.debug("Tickets: " + tickets);
+        tickets.sort(ticketComparatorByUserEmail());
+        LOG.debug("Sorted tickets: " + tickets);
+        return listByPage(tickets, pageSize, pageNum);
+    }
+
+    private List<Ticket> getBookedTickets(long eventId) {
+        return ticketService.retrieveTicketsByEvent(eventId);
+    }
+
+    public boolean cancelTicket(long ticketId) {
+        return ticketService.cancelBooking(ticketId);
+    }
+
+    private Comparator<Ticket> ticketComparatorByEventDate() {
+        return Comparator.comparing(o -> eventService.retrieveEventById(o.getEventId()).getDate());
+    }
+
+    private Comparator<Ticket> ticketComparatorByUserEmail() {
+        return Comparator.comparing(o -> userService.retrieveUserById(o.getUserId()).getEmail());
+    }
+
+    private void checkPageSize(int pageSize, int pageNum) {
```

Comment:
from name not clear check for what this checkers

isPageSizeIsPositiveOrThrowException(int pageSize, int pageNum) more readable